ncp-web/ncp-launcher.php

```diff
@@ -80,13 +80,33 @@
   }
 
   // launch
-  echo '{ "token": "' . getCSRFToken() . '",';     // Get new token
-  echo ' "ref": "' . $ncp_app          . '",';
-  echo ' "output": "" , ';
-  echo ' "ret": ';
 
-  exec( 'bash -c "sudo /home/www/ncp-launcher.sh ' . $ncp_app . '"' , $output , $ret );
-  echo '"' . $ret . '" }';
+  $ret = null;
+  $output = null;
```

Comment:
```suggestion
  $output = "";
```

Reply:
Hm, are you sure about that? I would consider `null` to be more accurate for an unset variable. I'm only referencing it here for context (maybe I can leave it out altogether - I'm more used to python's context management?)

Reply:
Fixed in cd8451d318815945bae42ddda0a16b0a1ff1468a